src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -245,15 +243,18 @@ std::tuple<float, float> activity_classifier::compute_validation_metrics(
 
     while (pending_batches.size() > remaining) {
 
+      size_t batch_num_correct = 0;
```

Comment:
Nit: Let's improve the spatial locality here by moving these variable definitions to just before they're used